### PR TITLE
Move phpcs checks from magento2 to magento-coding-standard repo

### PR DESCRIPTION
**Pull request by loginesta**
https://jira.corp.magento.com/browse/AC-201 Move phpcs checks from magento2 to magento-coding-standard repo

### CODE REVIEW

File: Magento2/ruleset.xml [hunk 1]
```diff
@@ -580,4 +580,13 @@
         <severity>5</severity>
         <type>warning</type>
     </rule>
+    <rule ref="Magento2.Annotation">
```
Comment (sivaschenko) [inline Magento2/ruleset.xml:583]: All the rules from the following files should be added as well:
 - https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/testsuite/Magento/Test/Html/_files/html/ruleset.xml
 - https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/testsuite/Magento/Test/Less/_files/lesscs/ruleset.xml
 - https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/framework/Magento/ruleset.xml

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @loginesta .

The remaining work would be:
 - Fix all the failing tests
 - Add all the rules from the rulesets specified in the comments

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @loginesta,
Could you also cover your change with tests?

File: Magento2/ruleset.xml (hunk 1 above)
Comment (loginesta) [reply Magento2/ruleset.xml:583]: These ones are already there, `EndFileNewline`, `LineEndings` and `DisallowTabIndent`.
Just adding the ones missing

**Comment by sivaschenko** [comment]
@ihor-sviziev we are moving a bulk of tests from magento2 repository to magento-testing-coverage (that is the first part). Considering the volume of tests to be moved, limited resources and missing test coverage on magento2 repository we would like to deffer test coverage for the Sniffs moved from magento2 in bulk.

File: Magento2/Sniffs/GraphQL/ValidArgumentNameSniff.php [hunk 2]
```diff
@@ -128,8 +128,7 @@ private function getArgumentListClosePointer($openParenthesisPointer, array $tok
     }
 
     /**
-     * Seeks the next available {@link T_OPEN_PARENTHESIS} token that comes directly after <var>$stackPointer</var>.
-     * token.
+     * Seeks the next available {@link T_OPEN_PARENTHESIS} token that comes directly after <var>$stackPointer</var> token.
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/GraphQL/ValidArgumentNameSniff.php:131]:
```suggestion
     * Find the argument list open pointer
     *
     * Seeks the next available {@link T_OPEN_PARENTHESIS} token
     * that comes directly after <var>$stackPointer</var> token.
```

**Review by ihor-sviziev — COMMENTED** [review]
You can try like this

File: Magento2/Sniffs/GraphQL/ValidEnumValueSniff.php [hunk 3]
```diff
@@ -68,8 +68,7 @@ public function process(File $phpcsFile, $stackPtr)
     }
 
     /**
-     * Seeks the next available token of type {@link T_CLOSE_CURLY_BRACKET} in <var>$tokens</var> and returns its
-     * pointer.
+     * Seeks the next available token of type {@link T_CLOSE_CURLY_BRACKET} in <var>$tokens</var> and returns its pointer.
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/GraphQL/ValidEnumValueSniff.php:71]:
```suggestion
     * Find the closing curly bracket pointer
     *
     * Seeks the next available token of type {@link T_CLOSE_CURLY_BRACKET} 
     * in <var>$tokens</var> and returns its pointer.
```

File: Magento2/Sniffs/GraphQL/ValidEnumValueSniff.php [hunk 4]
```diff
@@ -81,10 +80,9 @@ private function getClosingCurlyBracketPointer($startPointer, array $tokens)
     }
 
     /**
-     * Seeks the next available token of type {@link T_OPEN_CURLY_BRACKET} in <var>$tokens</var> and returns its
-     * pointer.
+     * Seeks the next available token of type {@link T_OPEN_CURLY_BRACKET} in <var>$tokens</var> and returns its pointer.
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/GraphQL/ValidEnumValueSniff.php:83]:
```suggestion
     * Find the opening curly bracket pointer
     *
     * Seeks the next available token of type {@link T_OPEN_CURLY_BRACKET}
     * in <var>$tokens</var> and returns its pointer.
```

**Review by ihor-sviziev — COMMENTED** [review]
And like this

**Review by sivaschenko — APPROVED** [review]
Test coverage and rules properties in ruleset.xml to follow

**Comment by sivaschenko** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by magento-engcom-team** [comment]
@sivaschenko the pull request successfully imported.

**Pull request merged**